src/cli/expand-patterns.js

```diff
@@ -65,9 +66,13 @@ async function* expandPatternsInternal(context) {
       continue;
     }
 
-    const stat = await statSafe(absolutePath);
+    const stat = await lstatSafe(absolutePath);
     if (stat) {
-      if (stat.isFile()) {
+      if (stat.isSymbolicLink()) {
+        yield {
+          error: `Explicitly specified ${pattern} is a symbolic link.`,
+        };
```

Comment:
It seems we will still expand other patterns, should we just throw?

Reply:
Dunno. We still expand other patterns in similar cases already:

```console
$ echo x > real.js
$ prettier -l nonexistent.js 'r*.js'
[error] No files matching the pattern were found: "nonexistent.js".
real.js
```

Reply:
That's because we don't exit if file not exits.  and we don't know if we can glob any file.

Reply:
Sorry, you are right..